selfservice/hook/password_migration_hook.go

```diff
@@ -0,0 +1,111 @@
+// Copyright © 2024 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package hook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/tidwall/gjson"
+	"go.opentelemetry.io/otel/codes"
+	semconv "go.opentelemetry.io/otel/semconv/v1.11.0"
+	"go.opentelemetry.io/otel/trace"
+	grpccodes "google.golang.org/grpc/codes"
+
+	"github.com/ory/herodot"
+	"github.com/ory/kratos/request"
+	"github.com/ory/kratos/schema"
+	"github.com/ory/x/otelx"
+)
+
+type (
+	PasswordMigration struct {
+		deps webHookDependencies
+		conf json.RawMessage
+	}
+	PasswordMigrationRequest struct {
+		Identifier string `json:"identifier"`
+		Password   string `json:"password"`
+	}
+	PasswordMigrationResponse struct {
+		Status string `json:"status"`
+	}
+)
+
+func NewPasswordMigrationHook(deps webHookDependencies, conf json.RawMessage) *PasswordMigration {
+	return &PasswordMigration{deps: deps, conf: conf}
+}
+
+func (p *PasswordMigration) Execute(ctx context.Context, data *PasswordMigrationRequest) (err error) {
+	var (
+		httpClient = p.deps.HTTPClient(ctx)
+		emitEvent  = gjson.GetBytes(p.conf, "emit_analytics_event").Bool() || !gjson.GetBytes(p.conf, "emit_analytics_event").Exists() // default true
```

Comment:
The default is already set to true in the config schema, I don't think that it is the best idea to re-implement that here. Just adds an additional (hidden) point where this is set.

Reply:
I am pretty sure that they do get applied, did you try? You can then also remove the other one.

Reply:
Yes, I tried, and it is indeed false when the config is empty.

Reply:
The reasons is that we do not parse the hook config as part of the koanf loading.